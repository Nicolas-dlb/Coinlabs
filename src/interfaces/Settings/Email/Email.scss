.settings_email {
  width: 85%;
  align-self: flex-end;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  padding: 20px;
  border-radius: 15px;
  background: var(--third-bg-color);
  margin-top: 2vh;
  .settings_email_input {
    display: flex;
    align-items: center;
    width: 100%;
    svg {
      width: 20px;
      height: 20px;
      fill: white;
      padding-left: 10px;
    }
    input {
      width: 100%;
    }
    button {
      margin-left: 15px;
      position: relative;
      top: 8px;
    }
  }
  .input {
    width: 60%;
  }
}

@media only screen and (min-width: 992px) {
  .settings_email {
    .input {
      width: 60%;
    }
  }
}
