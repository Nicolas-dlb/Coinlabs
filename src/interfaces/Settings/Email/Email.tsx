/* eslint-disable no-unused-vars */

import React, { useEffect, useState } from "react";
import { useDispatch } from "react-redux";
import { setUserEmail } from "redux/reducers/userSlice";
import "./Email.scss";
import $ from "jquery";
import { auth, db } from "firebaseConfig";

function Email() {
  const dispatch = useDispatch();
  const [email, setEmail] = useState("");

  const changeEmail = () => {
    const user = auth.currentUser!;
    const newEmail: any = $("#set_email").val();
    user
      .updateEmail(newEmail)
      .then(() => {
        // Update successful
        // ...
      })
      .catch((error) => {
        // An error occurred
        // ...
      });
    db.collection("users").doc(auth.currentUser?.uid).set(
      {
        email: newEmail,
      },
      { merge: true }
    );
    dispatch(setUserEmail(newEmail));
    setUserEmail(newEmail);
    $("#set_email").val("");
  };

  // const updateEmail = (newEmail: string) => {
  //   db.collection("users")
  //     .doc(auth.currentUser!.uid)
  //     .set({ email: newEmail }, { merge: true });
  // };

  // useEffect(() => {
  //   updateEmail(email);
  // }, [email]);
  return (
    <div className="settings_email">
      <h3>Email</h3>
      <div className="settings_email_input">
        <div className="input input_settings">
          <svg
            xmlns="http://www.w3.org/2000/svg"
            width="489"
            height="489"
            viewBox="0 0 489.2 489.2"
            className="email_icon"
          >
            <polygon points="0 386.6 143.5 214.9 0 99.6 " />
            <polygon points="489.2 386.6 489.2 99.6 345.6 214.9 " />
            <polygon points="487.9 398.7 489.2 398.7 489.2 397.7 " />
            <polygon points="0 398.7 1.2 398.7 0 397.7 " />
            <polygon points="480.8 398.7 479.9 397.7 334.6 223.7 244.6 296.1 154.6 223.7 9.2 397.7 8.4 398.7 " />
            <polygon points="152.6 204 158.1 208.5 163.6 212.9 244.6 277.9 325.5 212.9 331 208.5 336.5 204 477.9 90.5 11.3 90.5 " />
          </svg>
          <input
            id="set_email"
            onChange={(e) => setEmail(e.target.value)}
            type="text"
            placeholder="Enter new email"
            onKeyPress={(e) => {
              if (e.key === "Enter") {
                changeEmail();
              }
            }}
          />
        </div>
        <button onClick={changeEmail} type="button" className="btn_settings">
          Update
        </button>
      </div>
    </div>
  );
}

export default Email;
