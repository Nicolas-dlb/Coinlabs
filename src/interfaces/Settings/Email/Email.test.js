import React from "react";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import Email from "./Email";

describe("Email", () => {
  let EmailComponent;
  beforeEach(async () => {
    await act(async () => {
      EmailComponent = mount(<Email />);
    });
  });
  it("should render correctly", () => {
    expect(EmailComponent).toMatchSnapshot();
  });
  it("should render Email title", () => {
    expect(EmailComponent.find("h3").text()).toBe("Email");
  });
  it("should render an icon", () => {
    const icon = (
      <svg
        xmlns="http://www.w3.org/2000/svg"
        width="489"
        height="489"
        viewBox="0 0 489.2 489.2"
        className="email_icon"
      >
        <polygon points="0 386.6 143.5 214.9 0 99.6 " />
        <polygon points="489.2 386.6 489.2 99.6 345.6 214.9 " />
        <polygon points="487.9 398.7 489.2 398.7 489.2 397.7 " />
        <polygon points="0 398.7 1.2 398.7 0 397.7 " />
        <polygon points="480.8 398.7 479.9 397.7 334.6 223.7 244.6 296.1 154.6 223.7 9.2 397.7 8.4 398.7 " />
        <polygon points="152.6 204 158.1 208.5 163.6 212.9 244.6 277.9 325.5 212.9 331 208.5 336.5 204 477.9 90.5 11.3 90.5 " />
      </svg>
    );
    expect(EmailComponent.containsMatchingElement(icon)).toBe(true);
  });

  it("should render an input", () => {
    expect(EmailComponent.find("input").length).toBe(1);
  });
  it("should render a button called Update", () => {
    expect(EmailComponent.find("button").text()).toBe("Update");
  });
});
