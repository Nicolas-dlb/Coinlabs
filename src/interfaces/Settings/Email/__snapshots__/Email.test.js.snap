// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email should render correctly 1`] = `
<Email>
  <div
    className="settings_email"
  >
    <h3>
      Email
    </h3>
    <div
      className="settings_email_input"
    >
      <div
        className="input input_settings"
      >
        <svg
          className="email_icon"
          height="489"
          viewBox="0 0 489.2 489.2"
          width="489"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="0 386.6 143.5 214.9 0 99.6 "
          />
          <polygon
            points="489.2 386.6 489.2 99.6 345.6 214.9 "
          />
          <polygon
            points="487.9 398.7 489.2 398.7 489.2 397.7 "
          />
          <polygon
            points="0 398.7 1.2 398.7 0 397.7 "
          />
          <polygon
            points="480.8 398.7 479.9 397.7 334.6 223.7 244.6 296.1 154.6 223.7 9.2 397.7 8.4 398.7 "
          />
          <polygon
            points="152.6 204 158.1 208.5 163.6 212.9 244.6 277.9 325.5 212.9 331 208.5 336.5 204 477.9 90.5 11.3 90.5 "
          />
        </svg>
        <input
          id="set_email"
          onChange={[Function]}
          placeholder="Enter new email"
          type="text"
        />
      </div>
      <button
        className="btn_settings"
        type="button"
      >
        Update
      </button>
    </div>
  </div>
</Email>
`;
